genesyscloud/resource_genesyscloud_routing_queue.go

```diff
@@ -1308,6 +1309,16 @@ func buildSdkDirectRouting(d *schema.ResourceData) *platformclientv2.Directrouti
 			UseAgentAddressOutbound: &messageUseAgentAddressOutbound,
 		}
 
+		if backupQueueID == "" || backupQueueID == NullValue {
+			return &platformclientv2.Directrouting{
+				CallMediaSettings:    callSettings,
+				EmailMediaSettings:   emailSettings,
+				MessageMediaSettings: messageSettings,
+				WaitForAgent:         &waitForAgent,
+				AgentWaitSeconds:     &agentWaitSeconds,
+			}
+		}
+
 		return &platformclientv2.Directrouting{
```

Comment:
If settingsMap["backup_queue_id"] could be nil, I would recommend defining the Directrouting object as a variable at the beginning, and setting the backup queue id like this:

```golang
if backUpQueueId, ok := settingsMap["backup_queue_id"].(string); ok && backUpQueueId != "" {
  directRouting.BackupQueueId = &backUpQueueId
}
```

It is more readable than having the two returns statements at the bottom, one having one more field than the other